abcclassroom/template.py

```diff
@@ -265,11 +324,17 @@ def create_extra_files(config, template_repo, assignment):
     """
     course_dir = cf.get_config_option(config, "course_directory", True)
     extra_path = Path(course_dir, "extra_files")
+
+    # TODO - implement helper for this part
+    files_to_ignore = cf.get_config_option(config, "files_to_ignore", True)
+
     if extra_path.is_dir():
         print("Copying extra files: ")
         for f in extra_path.iterdir():
-            print(" {}".format(f.relative_to(course_dir)))
-            shutil.copy(f, template_repo)
+            if not ntpath.basename(f) in files_to_ignore:
```

Comment:
here `ntpath` supposedly ensures that this will work on windows as well... not tested but i read that was better than using os.path to check `basename`. open to any feedback

Reply:
If we ensure that f is a pathlib object, and not a string, you can use `name`  https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name and be safe on all platforms

Reply:
oh! ok i'll try `name`. we should be working with pathlib objects here.. i'm pretty sure i made sure that was the case. i went down a SO rabbit hole on paths and odd quirks on linux and windows... maybe i went too far. :)

Reply:
this one is really tricky. Pathlib.glob() method returns a generator object which in ptyhon you iterate through it once and then it's empty if you try to use it again. you can call list() on it and it will become a list b ut then you can't use the methods pathlib provides. but something odd i s happening when you iterate through the generator  because it seems like it's producing strings each time rather than a path object which means i can't use `.name` will revisit this as well in the future. it seems to be working now as i wrote it but worth a note here.

Reply:
is there any reason to not just use `shutil.copytree` here? with the ignore option? that seems like it will handle everything that we want it to handle even directories and cut back on code